lib/sunrise/sunrise.php

```diff
@@ -67,12 +67,18 @@ function handle_not_found_error( $error_type ) {
 	$is_web_request = Context::is_web_request();
 	if ( $is_web_request ) {
 		$is_maintenance_mode = Context::is_maintenance_mode();
+		$is_overdue_locked   = Context::is_overdue_locked();
 		if ( $is_maintenance_mode ) {
 			// 503 prevents page from being cached.
 			// We handle healthchecks earlier and don't have to worry about them.
 			$status_code = 503;
 			header( 'X-VIP-Go-Maintenance: true' );
 			$error_doc = sprintf( '%s/mu-plugins/errors/site-maintenance.html', WP_CONTENT_DIR );
+		} elseif ( $is_overdue_locked && Context::is_vip_env() ) {
+			// Similar with maintenance mode, but with a different message.
+			$status_code = 503;
```

Comment:
503 doesn't seem like the right status code, and will probably end up setting off various alarms and polluting metrics.

There's an "unofficial" 402 code for Payment Required that sounds right for this https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402

Reply:
Will 402s be cached in Varnish? Will need to look into how we handle that (e.g. if cached, we probably want to flush the cache after the lockout is removed)

Reply:
Replied in another spot, but I couldn't find any exceptions in varnish.vcl that would affect 402, so AFAIK it will receive the normal cache rules, which means we will need to do a cache flush on lockout...and on lockout for that matter.

Reply:
That's a very good point, and thanks for checking @nickdaugherty! I'm going to add the extra step for the cache flush on the respective GOOP endpoints.